File: part3/script/python/simple/hello_client.py
# Apache Thrift Hello World Python Client

import sys
sys.path.append("gen-py")
from hello import helloSvc

from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

trans = TSocket.TSocket("localhost", 9090)
trans = TTransport.TBufferedTransport(trans)
proto = TBinaryProtocol.TBinaryProtocol(trans)
client = helloSvc.Client(proto)

trans.open()
msg = client.getMessage("world")
print("[Client] received: %s" % msg)
trans.close()

File: part3/java/simple_mvn/src/main/java/HelloClient.java
// Apache Thrift Hello World Java Client

import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.TException;

public class HelloClient {

    public static void main(String[] args) {
        TSocket trans = new TSocket("localhost", 9090);
        TBinaryProtocol proto = new TBinaryProtocol(trans);
        helloSvc.Client client = new helloSvc.Client(proto);

        try {
            trans.open();
            String str = client.getMessage("world");
            System.out.println("[Client] received: " + str);
        } catch (TException ex) {
            System.out.println("Error: " + ex.toString());
        }
        trans.close();
    }
}
